### DISCUSSION TITLE
Example 16 control via Variable instead of external button

### DISCUSSION THREAD

**Discussion by chrwh** [HomeSpan/HomeSpan#740] [Q&A] [answered]
Hi,
I'm trying to change example 16 to be able to change the switches via an Variables in the main loop instead of external buttons.
Anybody know how to do this, and can show an example?
Thanks and best regards,
Christian

**Comment by HomeSpan** [comment, marked as answer]
There are various ways of doing this - the best way will depend on exactly what you are trying to do.  

For example, if there are conditions under which you want to turn on a specific switch, and those conditions are based only upon that one switch, you can simply include code for your logic in the `loop()` method of the specific Service of interest.

On the other hand, if you have something more generic you need to do that involves logic spanning multiple Services, it may be more appropriate to add this to the main Arduino `loop()`, which means you need to be able to *access those variables outside of their class*.  This is easily done if you create **global** variables  that point to Services you will create inside the `setup()` function.  For example, below is some code that is a variation of Example 5 showing how two LEDs can be referenced in the main Arduino `loop()`.  Note how we save the name of the two `DEV_LED()` objects using globally-scoped pointers defined at the top of the sketch.  In this simple example, the sketch automatically turns off both lights if it ever finds that both are on at the same time:

```C++

#include "HomeSpan.h" 
#include "DEV_LED.h"

DEV_LED *led16;     // create global pointers to the DEV_LED Service
DEV_LED *led17;

void setup() {

  Serial.begin(115200);

  homeSpan.begin(Category::Lighting,"HomeSpan LED");
  
  new SpanAccessory();  
    new Service::AccessoryInformation(); 
      new Characteristic::Identify();                
    led16=new DEV_LED(16);                  // save the name using a global pointer


  new SpanAccessory();   
    new Service::AccessoryInformation();    
      new Characteristic::Identify();                       
    led17=new DEV_LED(17);                  // save the name using a global pointer

}

//////////////////////////////////////

void loop(){
  
  homeSpan.poll();

  if(led16->power->getVal() && led17->power->getVal()){
    Serial.printf("Power overload!  Can't have both LED's on at the same time.  Turn off both LEDs...\n");
    led16->power->setVal(false);
    led17->power->setVal(false);
  }
  
}
```

**Reply by chrwh** [reply]
Thanks that was what I needed - it seems to be working :-)
